src/domain/notification/aggregate.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable max-lines-per-function */
+import type { StoreApi } from 'zustand'
+import { immer } from 'zustand/middleware/immer'
+import { createStore } from 'zustand/vanilla'
+import { devtools } from 'zustand/middleware'
+import type { IO } from 'fp-ts/lib/IO'
+import type { Reader } from 'fp-ts/lib/Reader'
+import * as A from 'fp-ts/lib/Array'
+import * as O from 'fp-ts/lib/Option'
+import { pipe } from 'fp-ts/lib/function'
+import type { ForgetType } from '@/util/type'
+import { eqNotification } from './entity'
+import type { NotificationID, NotificationType, Notifications } from './entity'
+
+type NotificationState<T> = {
+  aggregate: Notifications<T>
```

Comment:
Yes, I'm not happy with this naming too. 

Well, to my eyes it's more the aggregate roots:

```suggestion
  aggregateRoots: Notifications<T>
```

a pointer to the list of all notifications which are actually aggregate roots.

Reply:
Yes, but when I come to read about DDD structures, the most part of time, the root object in an aggregate is an entity: 

_"An Aggregate is a set of business objects (Value Objects or Entities) linked together. Among these objects, only one (in general, an Entity) will have a particular role: the role of root of the Aggregate. Any changes made to the aggregate by the rest of the code will have to go through a root method. It is therefore forbidden to modify the aggregate by directly modifying an unrooted object."_

So the notion of `aggregate...` looks a bit confusing for me for the naming of this entity...

Reply:
I don't clearly understand the point. `Notifications` is the agregate root (but not an entity), since it maintains the link between all the objects, which are entities (`Notification`).

We could have define the aggregate root that way:

```ts
export type Notifications<T> = Notification<T>[] & AggregateRoot
```

and the `State` is actually a "repository" - a place where all the aggregate roots are kept.

If it is confusing, we can simply name it "data":

```suggestion
  data: Notifications<T>
```

Reply:
Sure @fredericvilcot. What I propose: I made some organizational evolutions on the PR #213 that is more advanced than this one. I propose a session of pair programming to act together a canonical model of domains that we will then apply on this PR, if it suits you.

FYI @lolottetheclash

Reply:
@ccamel @fredericvilcot do we apply these updates in a next iteration?

Reply:
@0xLodz 
Yes, I'll suggest to leave it like that and immediately do a PR that will normalize these elements.